ocp_utilities/infra.py

```diff
@@ -81,6 +85,46 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure
+        MemoryPressure, PIDPressure, NetworkUnavailable condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes_with_conditions = {}
+    for node in nodes:
+        condition_type_list = []
+        for condition in node.instance.status.conditions:
+            if (
+                condition.type != Node.Condition.READY
+                and condition.status == Node.Condition.Status.TRUE
+            ):
+                condition_type_list.append(condition.type)
+
+        if condition_type_list:
+            unhealthy_nodes_with_conditions[node.name] = condition_type_list.copy()
```

Comment:
`condition_type_list.copy()`, why do you need to use copy?

Reply:
Agreed with @myakove, copy is redundant here. When each node's conditions are checked, the list variable holding these elements will be reset.